mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1497,21 +1511,49 @@ def _get_sqlalchemy_filter_clauses(parsed, session, dialect):
                 entity = SqlParam
             elif SearchUtils.is_tag(key_type, comparator):
                 entity = SqlTag
+            elif SearchUtils.is_dataset(key_type, comparator):
+                entity = SqlDataset
             else:
                 raise MlflowException(
                     "Invalid search expression type '%s'" % key_type,
                     error_code=INVALID_PARAMETER_VALUE,
                 )
 
-            key_filter = SearchUtils.get_sql_comparison_func("=", dialect)(entity.key, key_name)
-            val_filter = SearchUtils.get_sql_comparison_func(comparator, dialect)(
-                entity.value, value
-            )
-            non_attribute_filters.append(
-                session.query(entity).filter(key_filter, val_filter).subquery()
-            )
+            if entity == SqlDataset:
+                if key_name == "context":
+                    dataset_filters.append(
+                        session.query(entity, SqlInput, SqlInputTag)
+                        .join(SqlInput, SqlInput.source_id == SqlDataset.dataset_uuid)
+                        .join(
+                            SqlInputTag,
+                            and_(
+                                SqlInputTag.input_uuid == SqlInput.input_uuid,
+                                SqlInputTag.name == MLFLOW_DATASET_CONTEXT,
+                                SqlInputTag.value == value,
+                            ),
+                        )
+                        .subquery()
+                    )
```

Comment:
Does this work for comparators other than `=`? Do we have to include `SearchUtils.get_sql_comparison_func` here too?

Can we also add tests for `IN`, `LIKE`, and `ILIKE` on the `context` tag in the file store and sqlalchemy store?